### ISSUE TITLE
Debian packaging

### ISSUE THREAD

**Issue by adrelanos** [subgraph/paxrat#1]
Seems like some Debian developers are interested to have paxrat packaged for Debian.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=605090#487

Do you have the Debian packaging files in the wings or plan to create in near or mid future?

**Comment by mckinney-subgraph** [comment]
Closed issues #4 and #5 in relation to this one. Is there anything else you need to package this?

Also, out of curiosity, will this be made into a Debian source package or just a binary package?

**Comment by adrelanos** [comment]
mckinney-subgraph:

> Is there anything else you need to package this?

As far as I know. No. We'll open issues for anything that comes up.

> Also, out of curiosity, will this be made into a Debian source package or just a binary package?

If it gets added to official Debian repositories, then they will be
having also a source package since they must have source packages for
all binary packages by Debian policy.

**Comment by adrelanos** [comment]
FYI:
We are currently writing a RFP (request for packaging) to be soon submitted to Debian.
(Anyone can RFP, request any software to be packages for Debian, anytime.)
The latest draft can be found here:
https://phabricator.whonix.org/T203#7911

**Comment by HulaHoopWhonix** [comment]
Submitted now. Thanks everybody who made this possible.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=810479

**Comment by adrelanos** [comment]
Do you have the Debian packaging files in the wings? Or plan to create in near or mid future? How to you compile/update this for subgraph os without packaging files?

**Comment by micahflee** [comment]
I just discovered this project and it looks super useful. I'm also interested in very simple build/install instructions so I can use it will waiting for the debian package (I have little go experience).

**Comment by ageis** [comment]
+1 I'd also be interested to see paxrat in Debian; I actually just e-mailed the maintainers of grsecurity-related packages to give some prodding about getting paxctld in there - any tool to make PaX flags persistent. According to the docs/README, it would seem that paxrat supports some use cases that paxctld doesn't, and these are the reason why Subgraph chose to write their own daemon instead of using/contributing to paxctld. Maybe this is the wrong issue, but I'm still curious about that, since I'm evaluating which to use.

**Comment by adrelanos** [comment]
paxctld - PaX flags maintenance daemon - binary packages
https://grsecurity.net/download.php

vs

paxrat - paxrat is a utility to set PaX flags on a set of binaries. 
https://github.com/subgraph/paxrat

What's the difference? paxctld already is available as `.deb` package. Therefore might be easier to get into Debian.

**Comment by dma** [comment]
We needed to be able to set and maintain ELF program header pax flags for when SGOS is used in live mode because of missing extended attribute support in the filesystem. Therefore Paxrat supports setting both ELF and xattr pax flags. Yes, you may be able to use paxctld.

**Comment by adrelanos** [comment]
I have placed a $ 200 USD bounty on this ticket on bountysource.

**Introduction**:

paxrat is a utility to set PaX flags on a set of binaries.

---

**TODO**:

Add Debian packaging files for paxrat. Get feedback from debian.org developers. Make sure to meet Debian requirements to have the package added to the official debian.org repository.

---

**Debian RFP (request for packaging)**:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=810479

---

**paxrat upstream ticket**:
https://github.com/subgraph/paxrat/issues/1

---

**Mirrored from**:
https://phabricator.whonix.org/T468

---

**On bountysource**:
https://www.bountysource.com/issues/29647889-debian-packaging

---

**Bounty too low?**:

1) Go to https://www.bountysource.com/issues/29647889-debian-packaging
2) Click on "Developers"
3) Click on "Get Started"
4) Select Status "Bounty too low"
5) Enter your offer and press "Save".

**Comment by adrelanos** [comment]
Actually, it has already been packaged. Well, there is a binary Debian package...

Add to `/etc/apt/sources.list`.

```
deb https://devrepo.subgraph.com/subgraph aaron main
```

```
apt-get download paxrat
```

I failed to find the subgraph os Debian source packages. Asked subgraph os developers about it:
https://secure-os.org/pipermail/desktops/2016-May/000118.html

**Comment by santiagorr** [comment]
Hi,

I am packaging paxrat in debian collab-maint git repository:
https://anonscm.debian.org/gitweb/?p=collab-maint/paxrat.git

Testing packages for unstable are available at:

```
deb https://people.debian.org/~santiago/debian santiago-unstable/
deb-src https://people.debian.org/~santiago/debian santiago-unstable/
```

Feedback is always welcome :-)

Cheers,
Santiago

**Comment by santiagorr** [comment]
paxrat has been accepted into debian unstable :-)

https://packages.qa.debian.org/p/paxrat/news/20161128T200010Z.html

**Comment by mckinney-subgraph** [comment]
Wow! :)

**Comment by adrelanos** [comment]
What is the status of this ticket?

Does subgraph use https://packages.debian.org/stretch/paxrat now?

**Comment by mckinney-subgraph** [comment]
No, we don't use the Debian package. It doesn't make sense to because we need our users to have PaX flag updates faster than the time it takes for a package to pass through the Debian release cycle. We can release something in a day whereas it may take weeks/months for new upstream versions to get into Stretch.

**Comment by santiagorr** [comment]
El 09/04/17 a las 07:11, mckinney-subgraph escribió:
>    No, we don't use the Debian package. It doesn't make sense to because we
>    need our users to have PaX flag updates faster than the time it takes for
>    a package to pass through the Debian release cycle. We can release
>    something in a day whereas it may take weeks/months for new upstream
>    versions to get into Stretch.

If it's only about flag updates, with the recent changes that closed #20,
it would be easier to have config files independent from the paxrat
binary and its .deb.

Once those changes will be part of the Debian packages (i.e. for
Buster), if you'd ever want to use it, an option would be to have a
specific package for flag configs.
